src/ir/properties.h

```diff
@@ -146,6 +146,11 @@ inline Expression* getSignExtValue(Expression* curr) {
   if (curr->type != Type::i32) {
     return nullptr;
   }
+  // if (auto* unary = curr->dynCast<Unary>()) {
+  //   if (unary->op == ExtendS8Int32 || unary->op == ExtendS16Int32) {
+  //     return unary->value;
+  //   }
+  // }
```

Comment:
if uncomment this part tests will fail with this assert error:

Assertion failed: (int(_id) == iAssertion failed: (int(_id) == int(T::SpecificId)), function casnt(T::SpecificId)), function cast, file ..../binaryen/src/wasm.h, lis/Github/binaryen/src/wasm.h, line 676.

Reply:
I think what happens is that the pattern matching on (new) line 157 is what this function used to accept. And so the callers assumed that pattern. We'd need to see how to modify the callers so they also accept the new thing you want to add here.

Reply:
Alright, I found a problem